core/src/actors/session.rs

```diff
@@ -125,11 +130,53 @@ impl WriteHandler<Error> for Session {}
     /// This is main event loop for client requests
-    fn handle(&mut self, msg: Request, _ctx: &mut Self::Context) {
+    fn handle(&mut self, msg: Request, ctx: &mut Self::Context) {
         // Handler different types of requests
         match msg {
             Request::Message(message) => {
                 debug!("Session {} received message: {:?}", self.address, message);
+                let decoded_msg = WitnetMessage::try_from(message.to_vec());
+                match decoded_msg {
+                    Err(err) => debug!("Error decoding message: {:?}", err),
+                    Ok(msg) => {
+                        match msg.kind {
+                            Command::GetPeers => {
+                                // Get peers manager address
+                                let peers_manager_addr = System::current()
+                                    .registry()
+                                    .get::<peers_manager::PeersManager>();
+
+                                // Start chain of actions
+                                peers_manager_addr
+                                    // Send GetPeer message to peers manager actor
+                                    // This returns a Request Future, representing an asynchronous message sending process
+                                    .send(peers_manager::GetPeers)
+                                    // Convert a normal future into an ActorFuture
+                                    .into_actor(self)
+                                    // Process the response from the peers manager
+                                    // This returns a FutureResult containing the socket address if present
+                                    .then(|res, act, ctx| {
+                                        match res {
+                                            Ok(Ok(addresses)) => {
+                                                debug!("Get peers successfully registered into the Peers Manager");
+                                                let peers_msg: Vec<u8> = WitnetMessage::build_peers(&addresses).into();
+                                                act.framed.write(Response::Message(peers_msg.into()));
+                                            },
+                                            _ => {
+                                                debug!("Get peers register into Peers Manager failed");
+                                                // FIXME(#72): a full stop of the session is not correct (unregister should
+                                                // be skipped)
+                                                ctx.stop();
+                                            }
+                                        }
+                                        actix::fut::ok(())
+                                    })
+                                    .wait(ctx);
+                            }
+                            _ => panic!(""),
```

Comment:
Don't know if panicking is the best option here. I would discuss this with the team or link this as well to issue #72 , to be discussed in the near future.